src/threading.ts

```diff
@@ -117,12 +118,33 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(f: Py_FrameObject): void {
```

Comment:
I can change ```throwException``` so it takes an exception object.
But the comparison between exception and exception handler type occurs only in ```COMPARE_OP``` opcode. So ```tryCatchException``` only finds an exception handler. If an exception handler is found, but its type is different from the thrown exception, the exception is re-raised in the ```COMPARE_OP``` opcode. That's why tryCatchException doesn't take an exception object.
```
try:
    raise NameError
except ArithmeticError:
```

Here when ```NameError``` is raised, the thread looks for any exception handler. It finds one and when it reaches ```COMPARE_OP```, it compares the types. It fails in the example above so the exception is thrown again.